webrtc-stats.html

```diff
@@ -1294,10 +1297,106 @@
+                <dfn><code>qualityLimitationReason</code></dfn> of type <span class=
+                "idlMemberType"><a>RTCQualityLimitationReason</a></span>
+              </dt>
+              <dd>
+                <p>
+                  Only valid for video. The current reason for limiting the resolution and/or
+                  framerate, or "none" if not limited.
+                </p>
+              </dd>
+              <dt>
+                <dfn><code>qualityLimitationDurations</code></dfn> of type <span class=
+                "idlMemberType"><a>record&lt;RTCQualityLimitationReason, double&gt;</a></span>
+              </dt>
+              <dd>
+                <p>
+                  A record of the total time, in seconds, that this stream has spent being
+                  resolution and/or framerate limited for the reason specified by the record's
+                  key.
+                </p>
+              </dd>
+              <dt>
+                <dfn><code>qualityTotalLimitationDuration</code></dfn> of type <span class=
+                "idlMemberType"><a>double</a></span>
+              </dt>
+              <dd>
+                <p>
+                  The total time, in seconds, that this stream has spent being resolution and/or
+                  framerate limited (that <code>qualityLimitationReason</code> has had a value
+                  other than "none").
+                </p>
+              </dd>
             </dl>
           </section>
         </div>
       </section>
+      <section>
+        <h3>
+          <dfn>RTCQualityLimitationReason</dfn> enum
+        </h3>
+        <div>
+          <pre class="idl">enum RTCQualityLimitationReason {
+            "none",
+            "cpu",
+            "bandwidth",
+            "other",
+          };</pre>
+          <table data-link-for="RTCQualityLimitationReason" data-dfn-for="RTCQualityLimitationReason" class="simple">
+            <tbody>
+              <tr>
+                <th colspan="2">
+                  Enumeration description
+                </th>
+              </tr>
+              <tr>
+                <td>
+                  <dfn><code>none</code></dfn>
+                </td>
+                <td>
+                  <p>
+                    The resolution and/or framerate is not limited.
+                  </p>
+                </td>
+              </tr>
+              <tr>
+                <td>
+                  <dfn><code>cpu</code></dfn>
+                </td>
+                <td>
+                  <p>
+                    The resolution and/or framerate is primarily limited due to CPU load.
+                  </p>
+                </td>
+              </tr>
+              <tr>
+                <td>
+                  <dfn><code>bandwidth</code></dfn>
+                </td>
+                <td>
+                  <p>
+                    The resolution and/or framerate is primarily limited due to by bandwidth
+                    or packet loss.
```

Comment:
The resolution and/or framerate is primarily limited due to congestion cues during bandwidth estimation.

Reply:
I like the above description. 

What would you think about expanding the dictionary and including bandwidth-interarrival, bandwidth-rtt,bandwith-loss, bandwidth-other in order to allow better classification of what piece of the network congestion control was enabled (and implementations could always just use just "bandwidth" if desired). For our customers, the thing they are most able to fix is network issues and having the type of network trouble clearer makes it much simpler to correct the right thing.

Reply:
Deciding why bandwidth estimation took the decision it did seems to me to be a chancy process. If we want to do that, I think we should instrument the bandwidth estimator (likely on the ICETransport object), not the video track.

Reply:
I used vr00ms description. If needed, I'd prefer more details to be surfaced through other stats.